### ISSUE TITLE
Add rule: prefer-object-spread

### ISSUE THREAD

**Issue by pertrai1** [pertrai1/eslint-plugin-llm-core#11]
## Problem

LLMs use Object.assign for shallow copies when spread is more idiomatic.

## Rule

Prefer `{...obj}` over Object.assign for shallow copies.

### ❌ Bad
```ts
const merged = Object.assign({}, defaults, options);
const copy = Object.assign({}, original);
```

### ✅ Good
```ts
const merged = { ...defaults, ...options };
const copy = { ...original };
```

## Why it matters

- Spread is more readable
- Spread is immutable by default
- Object.assign can accidentally mutate
- Modern convention
